### PR TITLE
Persistent memory

### PR DESCRIPTION
**Pull request by pstills**
I would like to merge this with main. it would be crazy if i was allowed to. it would be chill and it would make the code better.

### CODE REVIEW

File: src/main.py [hunk 1]
```diff
@@ -151,7 +72,7 @@ async def on_message(message: DiscordMessage):
         conversation_open = False
     print(f"Time Diff: {timeDif.seconds}")
     conversation_open_time = datetime.datetime.now()
-    if (message.channel.id == 1077110958738112614) and activationString.lower() in message.content.lower() and replyChance < 101:
+    if (message.channel.id == 1077110958738112614 or message.channel.id == 1079331067124785203) and activationString.lower() in message.content.lower() and replyChance < 101:
```
[the remark this answers is not in the document]
Comment (pstills) [reply src/main.py:75]: how though?

File: embedder.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import os
+import pickle
+
+testPath = '/Users/philipstilwell/Downloads/icecreammansample.docx'
+realPath = '/Users/philipstilwell/Downloads/icecreammandoc.docx'
+embeddingPath = '/Users/philipstilwell/Downloads/fulldaxbibliography_data.pkl'
+model = SentenceTransformer('all-mpnet-base-v2')
+embedding_size = 768
+
+class SemanticRecallModule:
```
[the remark this answers is not in the document]
Comment (pstills) [reply embedder.py:15]: ohhhh.

File: embedder.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+            self.__data += pickle.load(open(embeddingPath, "rb"))['sentences']
+        else:
+            embeddings = model.encode(data, show_progress_bar=True)
+            self.__data += data
+            sentences = data
+            with open(embeddingPath, "wb") as fOut:
+                pickle.dump({'sentences': sentences, 'embeddings': embeddings}, fOut)
+        self.__index.add(embeddings)
+        
+
```
[the remark this answers is not in the document]
Comment (pstills) [reply embedder.py:35]: ok ?

File: src/embedding_recall.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import docx
+import re
+import os
+import pickle
+
+embeddingPath = 'src/fulldaxbibliography_data.pkl'
+model = SentenceTransformer('all-mpnet-base-v2')
+embedding_size = 768
+
+class SemanticRecallModule:
```
[the remark this answers is not in the document]
Comment (pstills) [reply src/embedding_recall.py:13]: no it woouldnt

**Comment by pstills** [comment]
> Looks good functionally, but could use a few style edits for readability

how wod you know you didn't even write it????

**Review by jacquesdebar — APPROVED** [review]
Looks great!

**Pull request merged**